apps/api/src/domain/domainModelValidators/contextValidators/validateEdgeConnection/index.ts

```diff
@@ -141,5 +147,12 @@ export default (input: unknown): Valid | InternalError => {
 
     if (contextModelInvariantErrors.length > 0) allErrors.push(...contextModelInvariantErrors);
 
+    /**
+     * Validate requirements for an identity connection, if that is what we have.
```

Comment:
This doesn't quite match the domain. What's important is the self \ dual connection distinction, which we have made explicit at the top-level. The top-level switch in the validation logic should be on this. Then the rule should be
- self -> A self connection cannot use the identity context
- dual -> If one member uses the identity context, both must